egov-searcher/src/main/java/org/egov/search/service/SearchService.java

```diff
@@ -56,19 +57,35 @@ public Object searchData(SearchRequest searchRequest, String moduleName, String
 		Definition searchDefinition = null;
 		searchDefinition = searchUtils.getSearchDefinition(searchDefinitionMap, moduleName, searchName);
 		List<String> maps = new ArrayList<>();
+		Object data = null;
 		try{
-			maps = searchRepository.fetchData(searchRequest, searchDefinition);
+			if(null != searchDefinition.getIsCustomerRowMapEnabled()) {
+				if(!searchDefinition.getIsCustomerRowMapEnabled()) {
+					maps = searchRepository.fetchData(searchRequest, searchDefinition);
+				}else {
+					//This is a custom logic for bill-genie, we'll need to write code seperately to support custom rowmap logic for any search.
+					data =  searchRepository.fetchWithCustomMapper(searchRequest, searchDefinition);
```

Comment:
Best to abstract this custom row mapper handling logic with an interface implementation than hard wiring the custom row mapper functionality to bill genie.

Reply:
sure, why not?